### PR TITLE
#1711 add tags to Ftp, Hadoop, JDBC, Node, Topic, Pipeline, Connector…

### PR DESCRIPTION
**Pull request by chia7712**
… and Object

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1711)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/BrokerApi.scala [hunk 1]
```diff
@@ -124,7 +125,7 @@ object BrokerApi {
 
   final class Access private[BrokerApi] extends ClusterAccess[BrokerClusterInfo](BROKER_PREFIX_PATH) {
     def request: Request = new Request {
-      private[this] var name: String = _
+      private[this] var name: String = CommonUtils.randomString(10)
```
Comment (saivirtue) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/BrokerApi.scala]: Our name limit is defined by `ZookeeperApi.LIMIT_OF_NAME_LENGTH`, is it more suitable here?

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala [hunk 2]
```diff
@@ -106,6 +113,7 @@ private[configurator] object PipelineRoute {
           case (pipeline, clustersOption) =>
             clustersOption
               .map { clusters =>
+                println(s"[CHIA] tags:${pipeline.tags}")
```
Comment (saivirtue) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: ?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ShabondiApi.scala [hunk 3]
```diff
@@ -36,6 +36,9 @@ final object ShabondiApi {
                                        instances: Int)
       extends Data {
     override def kind: String = "shabondi"
+
+    // TODO: Does shabondi need the tags? by chia
+    override def tags: Set[String] = Set.empty
```
Comment (saivirtue) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ShabondiApi.scala]: what is this TODO mean?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala [hunk 4]
```diff
@@ -77,6 +77,9 @@ object StreamApi {
                                         lastModified: Long)
       extends Data {
     override def kind: String = "streamApp"
+
+    // TODO: see https://github.com/oharastream/ohara/issues/1544
+    override def tags: Set[String] = Set.empty
```
Comment (saivirtue) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala]: Could you leave some details about the TODO?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/BrokerApi.scala (hunk 1 above)
Comment (chia7712) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/BrokerApi.scala]: You are right.  I did those changes by IDE so most code are copy-and-paste :(

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala (hunk 2 above)
Comment (chia7712) [reply ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: my bad :(

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala (hunk 4 above)
Comment (chia7712) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala]: I neglected this one :(

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/ConnectorRoute.scala [hunk 5]
```diff
@@ -107,13 +106,13 @@ private[configurator] object ConnectorRoute extends SprayJsonSupport {
...
             // Update request may not carry the name via payload so we copy the name from url to payload manually
-            Map(SettingDefinition.CONNECTOR_NAME_DEFINITION.key() -> JsString(name))
-        )
-        CollieUtils.workerClient(updatedRequest.workerClusterName).flatMap {
+            Map(SettingDefinition.CONNECTOR_NAME_DEFINITION.key() -> JsString(name)))
+
+        CollieUtils.workerClient(newSettings.workerClusterName).flatMap {
           case (cluster, wkClient) =>
             wkClient.exist(name).map {
               if (_) throw new IllegalArgumentException(s"connector:$name is not stopped")
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/ConnectorRoute.scala]: Would rephrase this to _Connector: $name is running_

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala [hunk 6]
```diff
@@ -89,7 +92,11 @@ private[configurator] object PipelineRoute {
                      "... please file a issue to fix this ... by chia")),
                objects = Seq.empty,
                workerClusterName = wkName,
-               lastModified = CommonUtils.current()
+               lastModified = CommonUtils.current(),
+               tags = request.tags.getOrElse(
+                 throw new NoSuchElementException(
+                   "We produced a bug here since we must fill the tags for all input updates before processing it " +
+                     "... please file a issue to fix this ... by chia")),
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: _an_ issue

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala [hunk 7]
```diff
@@ -192,7 +201,8 @@ private[configurator] object PipelineRoute {
                       error = Some(s"Failed to get status and type of connector:${data.name}." +
                         s"This may be temporary since our worker cluster is too busy to sync status of connector. ${e.getMessage}"),
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: reads better to me :)

_This could be a temporary issue since our..._

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala [hunk 8]
```diff
@@ -222,7 +233,8 @@ private[configurator] object PipelineRoute {
                       error = Some(s"Failed to get status of streamApp: ${data.name}." +
                         s"This may be temporary since our container cluster is too busy to sync status of streamApp. ${e.getMessage}"),
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: Same here

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala [hunk 9]
```diff
@@ -388,7 +403,10 @@ private[configurator] object PipelineRoute {
             meterCache: MeterCache): server.Route =
     RouteUtils.basicRoute[Creation, Update, Pipeline](
       root = PIPELINES_PREFIX_PATH,
-      hookOfCreation = (creation: Creation) => assertNoUnknown(creation).flatMap(toRes(_)),
+      hookOfCreation = (creation: Creation) => {
+        println(s"[CHIA] creation v0 tags:${creation.tags}")
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/PipelineRoute.scala]: Same as @saivirtue pointed out in the above comment :)

**Comment by chia7712** [comment]
unrelated error

**Comment by jackyoh** [comment]
retry configurator

**Comment by chia7712** [comment]
unrelated error
